quantlib/pricingengines/bondfunctions.pyx

```diff
@@ -0,0 +1,162 @@
+include '../types.pxi'
+
+#QL imports
+from quantlib.instruments._bonds cimport Bond as QLBond
+from quantlib.time._date cimport Day, Month, Year, Date as QLDate
+from quantlib.time._period cimport Frequency as _Frequency
+from quantlib.time._daycounter cimport DayCounter as _DayCounter
+from quantlib._interest_rate cimport InterestRate as _InterestRate
+cimport quantlib.termstructures._yield_term_structure as _yt
+cimport quantlib.pricingengines._bondfunctions as _bf
+
+#pyql imports
+from quantlib.handle cimport shared_ptr
+from cython.operator cimport dereference as deref
+from quantlib.interest_rate cimport InterestRate
+from quantlib.instruments.bonds cimport Bond
+from quantlib.time.date cimport date_from_qldate, Date
+from quantlib.termstructures.yields.yield_term_structure cimport YieldTermStructure
+from quantlib.time.daycounter cimport DayCounter
+
+cimport quantlib.pricingengines.bondfunctions
+cimport quantlib.time._date as _dt
+
+#cdef _bonds.Bond* get_bond(Bond bond):
+#    """ Utility function to extract a properly casted Bond pointer out of the
+#    internal _thisptr attribute of the Instrument base class. """
+#
+#    cdef _bonds.Bond* ref = <_bonds.Bond*>bond._thisptr.get()
+cdef extern from 'ql/compounding.hpp' namespace 'QuantLib':
```

Comment:
those cdef's are duplicated from the ones in `_bondfunctions.pxd`. Do you really need them?